src/nvim/tui/tui.c

```diff
@@ -542,18 +545,26 @@ void tui_io_driven_loop(void **argv){
     loop_poll_events(&main_loop, -1);  // tui_loop.events is never processed
     // LOOP_PROCESS_EVENTS(&main_loop, main_loop.events, 0);
   }
-
   TUIData *data = ui->data;
   tinput_destroy(&data->input);
   signal_watcher_stop(&data->cont_handle);
   signal_watcher_close(&data->cont_handle, NULL);
   signal_watcher_close(&data->winch_handle, NULL);
+}
+
+void tui_data_destroy(void **argv) {
+  UI *ui = argv[0];
+  TUIData *data = ui->data;
   kv_destroy(data->invalid_regions);
   kv_destroy(data->attrs);
   xfree(data->space_buf);
   xfree(data);
 }
 
+argv_callback tui_get_data_destroy_callback(void) {
```

Comment:
not needed, `main.c` can just use `tui_data_destroy` directly. (same for the two `callback` functions below)